src/EventListener/MessengerListener.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Adtechpotok\Bundle\SymfonyOpenTracing\EventListener;
+
+use Adtechpotok\Bundle\SymfonyOpenTracing\Contract\GetSpanNameByMessage;
+use OpenTracing\Tracer;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageHandledEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageReceivedEvent;
+
+class MessengerListener
+{
+    /**
+     * @var Tracer
+     */
+    protected $tracer;
+
+    /**
+     * @var GetSpanNameByMessage
+     */
+    private $nameGetter;
+
+    public function __construct(Tracer $tracer, GetSpanNameByMessage $nameGetter)
+    {
+        $this->tracer = $tracer;
+        $this->nameGetter = $nameGetter;
+    }
+
+    public function onWorkerMessageReceived(WorkerMessageReceivedEvent $event)
+    {
+        $envelope = $event->getEnvelope();
+        $name = $this->nameGetter->getNameByMessage($envelope);
+        $this->tracer->startActiveSpan($name);
+        $this->tracer->getActiveSpan()->setTag('messenger.message', get_class($envelope->getMessage()));
```

Comment:
```suggestion
        $this->tracer->startActiveSpan($name, [
            'tags' => ['messenger.message' => \get_class($envelope->getMessage()],
        ]);
```

I'm not 100% sure but this should do the same job.

Reply:
yeap, I've combined both method calls